js/post/update_post.js

```diff
@@ -223,82 +118,76 @@ function updateCategory(category, status) {
 /**
  * Updates Credit Count for each POSt.
  **/
+ 
 function fillCreditCount() {
     'use strict';
 
-    specCount = creditCountSpec  + creditCount300and400.spec;
-    majCount = creditCountMaj + creditCount300and400.maj;
-    minCount = creditCountMin + creditCount300and400.min;
-
-
-    fillSpecCreditCount(specCount);
-    fillMajCreditCount(majCount);
-    fillMinCreditCount(minCount);
+    fillSpecCreditCount();
+    fillMajCreditCount();
+    fillMinCreditCount();
 }
 
 
+
 /**
  * Autofills the credit count for specialist
- * @param {number} specCount The credit count for specialist
 **/
-function fillSpecCreditCount(specCount) {
+function fillSpecCreditCount() {
     'use strict';
 
-    if (specCount >= 12) {
+    if (specialist.creditCount >= 12) {
         $('#spec_creds').html('(12.0/12.0)');
     } else {
-        $('#spec_creds').html('(' + (specCount).toFixed(1) + '/12.0)');
+        $('#spec_creds').html('(' + specialist.creditCount.toFixed(1) + '/12.0)');
     }
 }
 
 
 /**
  * Autofills the credit count for major
- * @param {number} majCount The credit count for major
 **/
-function fillMajCreditCount(majCount) {
+function fillMajCreditCount() {
     'use strict';
 
-    if (majCount >= 8) {
+    if (major.creditCount >= 8) {
         $('#maj_creds').html('(8.0/8.0)')
     } else {
-        $('#maj_creds').html('(' + (majCount).toFixed(1) + '/8.0)');
+        $('#maj_creds').html('(' + major.creditCount.toFixed(1) + '/8.0)');
     }
 }
 
 
 /**
  * Autofills the credit count for minor.
- * @param {number} minCount The credit count for minor
 **/
-function fillMinCreditCount(minCount) {
+function fillMinCreditCount() {
     'use strict';
 
-    if (minCount >= 4) {
+    if (minor.creditCount >= 4) {
         $('#min_creds').html('(4.0/4.0)')
     } else {
-        $('#min_creds').html('(' + (minCount).toFixed(1) + '/4.0)');
+        $('#min_creds').html('(' + minor.creditCount.toFixed(1) + '/4.0)');
     }
 }
 
 
 /**
  * Autofills extra 200-level courses for last minor constraint.
- * @param {number} index The textbox number we are at
- * @param {HTMLElement} min300s Array of textbox elements to fill
 **/
-function addExtraMinCourses(index, min300s) {
+
+function addExtraMinCourses() {
     'use strict';
 
+    var minExtra = $('#minextra')[0].getElementsByTagName('input');
+
     for (var m = 0; m < 3; m++) {
-        if (index === 3) {
+        if (minor.filledTextboxesExtra === minor.textboxesExtra) {
```

Comment:
I feel like it would kind of end up being the same thing, unless I'm missing something. The for loop is to loop over the additionalMin200 array in the minor object, not to loop over the textboxes.